tests/operators/test_einsum_op.py

```diff
@@ -0,0 +1,47 @@
+"""Tests for Zero Pad Operator class."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+import torch
+from mrpro.operators._EinsumOp import EinsumOp
+
+from tests import RandomGenerator
+from tests.helper import dotproduct_adjointness_test
+
+
+@pytest.mark.parametrize('dtype', ['float32', 'complex128'])
+@pytest.mark.parametrize(
+    ('a_shape', 'x_shape', 'rule', 'y_shape'),
+    [
+        ((3, 3), (3), 'ij,j->i', (3)),
+        ((2, 4, 3), (2, 3), '...ij, ...j->...i', (2, 4)),
+        ((4, 3), (3, 2), '... i j , j k -> i k', (4, 2)),
```

Comment:
consider including further to cover other operations as well, e.g. Hadamard product etc
see https://rockt.github.io/2018/04/30/einsum